### PR TITLE
[avfoundation] Fix AVAudioUnitComponentManager default .ctor crash

### PR DESCRIPTION
**Pull request by VincentDondain**
It looks like this should not have been bound since there's a singleton-like static property to use the type.
Keeping binary compatibility.

- Fixes github issue #3303: AVAudioUnitComponentManager default .ctor can crash on macOS 10.13.3
(https://github.com/xamarin/xamarin-macios/issues/3303)

### CODE REVIEW

File: src/AVFoundation/AVAudioUnitComponentManager.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace XamCore.AVFoundation {
+
+#if !WATCH
+	public partial class AVAudioUnitComponentManager {
+#if !XAMCORE_4_0
+		[Obsolete ("Please use the static 'SharedInstance' property as this type is not meant to be created.")]
+		[CompilerGenerated]
+		[EditorBrowsable (EditorBrowsableState.Never)]
+		[Export ("init")]
+		public AVAudioUnitComponentManager () : base (NSObjectFlag.Empty)
```
Comment (dalexsoto) [inline src/AVFoundation/AVAudioUnitComponentManager.cs]: I think you can inline this directly in the ApiDefinition so no need for an additional manual code file 

```
[Obsolete ("Please use the static 'SharedInstance' property as this type is not meant to be created.")]
[Export ("init")]
IntPtr Constructor ();
```

File: src/AVFoundation/AVAudioUnitComponentManager.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		{
+			if (IsDirectBinding) {
+				Handle = XamCore.ObjCRuntime.Messaging.IntPtr_objc_msgSend (this.Handle, Selector.GetHandle ("init"));
+			} else {
+				Handle = XamCore.ObjCRuntime.Messaging.IntPtr_objc_msgSendSuper (this.SuperHandle, Selector.GetHandle ("init"));
```
Comment (spouliot) [inline src/AVFoundation/AVAudioUnitComponentManager.cs]: If you copy/paste the generated code then you'll end up with the same crash when running tests ;-)

File: src/AVFoundation/AVAudioUnitComponentManager.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.AVFoundation {
+
+#if !WATCH
+	public partial class AVAudioUnitComponentManager {
+#if !XAMCORE_4_0
+		[Obsolete ("Please use the static 'SharedInstance' property as this type is not meant to be created.")]
+		[CompilerGenerated]
```
Comment (spouliot) [inline src/AVFoundation/AVAudioUnitComponentManager.cs]: ^ remove that

File: src/AVFoundation/AVAudioUnitComponentManager.cs (hunk 2 above)
Comment (VincentDondain) [reply src/AVFoundation/AVAudioUnitComponentManager.cs]: I thought we ignored Obsoletes (:

Oops

Comment (spouliot) [reply src/AVFoundation/AVAudioUnitComponentManager.cs]: we _might_ be but there's a bug open for that so I'm not sure

**Comment by VincentDondain** [comment]
@spouliot looks you hit the comment button (so close :P). Can you "approve approve" so that it dismisses your request for changes?

**Comment by spouliot** [comment]
xammac failure is a random issue -> https://github.com/xamarin/maccore/issues/624

**Comment by spouliot** [comment]
That also does not fix everything but it's another small step in that direction

**Pull request merged**